File: hometask7/src/main/java/com/example/hometask7/ItemAdapter.java
package com.example.hometask7;

import android.os.Handler;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.hometask7.DAO.ContactDao;
import com.example.hometask7.data_base.MyDataBase;
import com.example.hometask7.entity.Contact;

import java.util.ArrayList;
import java.util.List;


public final class ItemAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private OnItemClickListener onItemClickListener;
    private Handler handler;
    private List<Contact> contactList = new ArrayList<>();

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public void setContactList(List<Contact> contactList) {
        this.contactList = contactList;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new RecyclerView.ViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(R.layout.item, parent, false)) {
        };
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        holder.itemView.setOnClickListener(v -> onItemClickListener.onClick(position));
        handler = new Handler();

            Contact contact = contactList.get(position);
            TextView name = holder.itemView.findViewById(R.id.contactNameReady);
            TextView emailOrNumber = holder.itemView.findViewById(R.id.contactEmailOrPhoneReady);
            ImageView imgV = holder.itemView.findViewById(R.id.imgButtonFromItem);

            name.post(() -> name.setText(contact.getName()));
            emailOrNumber.post(() -> emailOrNumber.setText(contact.getEmailOrPhone()));
            imgV.post(() -> imgV.setImageResource(contact.getSrc()));
        };


    @Override
    public int getItemCount() {
       return contactList != null ? contactList.size() : 0;
    }

    public interface OnItemClickListener {
        void onClick(int position);
    }

    public void deleteItem(Contact contact, ContactDao contactDao, int position, int contactsSize) {
        Thread thread = new Thread(new Runnable() {
            @Override
            public void run() {
                contactDao.delete(contact);
            }
        });
        notifyItemRemoved(position);
        notifyItemRangeRemoved(position, contactsSize);
    }

}

File: ht4/src/main/java/com/example/ht4/ItemAdapter.kt
package com.example.ht4

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item.view.*

class ItemAdapter(private var context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
            return object : RecyclerView.ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false))
            {}
        }

        override fun getItemCount(): Int {
            return Store.instance.size()
        }

        override fun onBindViewHolder(holder: RecyclerView.ViewHolder, index: Int) {
            holder.itemView.setOnClickListener{v: View? ->
                val intent = Intent(context, EditContact::class.java)
                intent.putExtra("index", index)
                context.startActivity(intent)
            }
            val item = Store.instance.get(index)
            holder.itemView.imgButtonFromItem.setImageResource(item.src)
            holder.itemView.contactNameReady.text = item.name
            holder.itemView.contactEmailOrPhoneReady.text = item.email
        }
    }